core/js/src/main/scala/com/avsystem/commons/serialization/nativejs/NativeFormatOptions.scala

```diff
@@ -0,0 +1,40 @@
+package com.avsystem.commons
+package serialization.nativejs
+
+import com.avsystem.commons.misc.{AbstractValueEnum, AbstractValueEnumCompanion, EnumCtx}
+
+/**
+  * Specifies format used by `NativeJsonOutput.writeLong` / `NativeJsonInput.readLong`
+  * to represent [[Long]]. JS does not support 64-bit representation.
+  */
+final class NativeLongFormat(implicit ctx: EnumCtx) extends AbstractValueEnum
+object NativeLongFormat extends AbstractValueEnumCompanion[NativeLongFormat] {
+  final val RawString: Value = new NativeLongFormat
+  final val JsNumber: Value = new NativeLongFormat
+  final val JsBigInt: Value = new NativeLongFormat
+}
+
+/**
+  * Specifies format used by `NativeJsonOutput.writeTimestamp` / `NativeJsonInput.readTimestamp`
+  * to represent timestamps.
+  */
+final class NativeDateFormat(implicit ctx: EnumCtx) extends AbstractValueEnum
+object NativeDateFormat extends AbstractValueEnumCompanion[NativeDateFormat] {
+  final val RawString: Value = new NativeDateFormat
+  final val JsNumber: Value = new NativeDateFormat
+  final val JsDate: Value = new NativeDateFormat
+}
+
```

Comment:
I don't have a usecase right now. Support for Hex binary format would require implementing writing/parsing Hex on SJS side. I think it is already done somewhere in JsonStringIn/Out, but larger refactor would be needed to use it. It can be added in the future I suppose